tkg/managementcomponents/management_component_install.go

```diff
@@ -290,14 +291,18 @@ func InstallManagementComponents(mcip *ManagementComponentsInstallOptions) error
 
 // InstallKappController installs kapp-controller to the cluster
 func InstallKappController(clusterClient clusterclient.Client, kappControllerOptions KappControllerOptions) error {
+	// Ensure last-applied annotation is present before kapp-controller upgrade
+	if err := clusterClient.GetResource(&appsv1.Deployment{}, constants.KappControllerDeploymentName, kappControllerOptions.KappControllerInstallNamespace, nil, nil); err == nil { // only attempt to add annotation when kapp-controller exists on the cluster
```

Comment:
What if an error happens here?

Reply:
Should we return error in case its other than NotFound and only call PatchKappControllerLastAppliedAnnotation if deployment is found ?

Reply:
If error happens, then we skip adding last-applied annotation on the `kapp-controller`. Because it might be `NotFound` error in the install phase, so this add annotation logic is not required

Reply:
> Should we return error in case its other than NotFound
I don't think it's necessary. If error other than `NotFound` is returned, during install phase, other unexpected error returned by `GetResource` will stop installing `kapp-controller`. We should continue to install `kapp-controller` regardless to these errors because they are not relevant to kapp-controller install.

Reply:
Let's ask @vijaykatam opinion on whether we can leave other unexpected errors unhandled. Otherwise the changes look good to me